src/MagnumPlugins/DdsImporter/DdsImporter.cpp

```diff
@@ -0,0 +1,387 @@
+            case DdsCompressionTypes::DXT5:
+                colorFormat.compressed = CompressedColorFormat::RGBAS3tcDxt5;
+                break;
+            default:
+                Error() << "unknown texture compression '" + fourcc(ddsh.ddspf.fourCC) + "'";
+        }
+        compressed = true;
+    } else if (ddsh.ddspf.rgbBitCount == 32 &&
+               ddsh.ddspf.rBitMask == 0x00FF0000 &&
+               ddsh.ddspf.gBitMask == 0x0000FF00 &&
+               ddsh.ddspf.bBitMask == 0x000000FF &&
+               ddsh.ddspf.aBitMask == 0xFF000000) {
+        colorFormat.uncompressed = ColorFormat::BGRA;
+    } else if (ddsh.ddspf.rgbBitCount == 32 &&
+               ddsh.ddspf.rBitMask == 0x000000FF &&
+               ddsh.ddspf.gBitMask == 0x0000FF00 &&
+               ddsh.ddspf.bBitMask == 0x00FF0000 &&
+               ddsh.ddspf.aBitMask == 0xFF000000) {
+        colorFormat.uncompressed = ColorFormat::RGBA;
+    } else if (ddsh.ddspf.rgbBitCount == 24 &&
+               ddsh.ddspf.rBitMask == 0x000000FF &&
+               ddsh.ddspf.gBitMask == 0x0000FF00 &&
+               ddsh.ddspf.bBitMask == 0x00FF0000) {
+        colorFormat.uncompressed = ColorFormat::RGB;
+        components = 3;
+    } else if (ddsh.ddspf.rgbBitCount == 24 &&
+               ddsh.ddspf.rBitMask == 0x00FF0000 &&
+               ddsh.ddspf.gBitMask == 0x0000FF00 &&
+               ddsh.ddspf.bBitMask == 0x000000FF) {
+        colorFormat.uncompressed = ColorFormat::BGR;
+        components = 3;
+    } else if (ddsh.ddspf.rgbBitCount == 8) {
+#ifndef MAGNUM_TARGET_GLES2
+        colorFormat.uncompressed = ColorFormat::Red;
+#else
+        colorFormat.uncompressed = ColorFormat::Luminance;
+#endif
+    } else {
+        Error() << "Unknow texture format";
+    }
+
+    const UnsignedInt width = ddsh.width;
+    const UnsignedInt height = ddsh.height;
+    const UnsignedInt depth = Math::max(ddsh.depth, 1u);   // set to 1 if 0
+
+    /* check how many mipmaps to load */
+    const UnsignedInt numMipmaps = (ddsh.flags & DdsDescriptionFlag::MipMapCount)
+                    ? ((ddsh.mipMapCount == 0) ? 0 : ddsh.mipMapCount - 1)
+                    : 0;
+
+    /* load all surfaces for the image (6 surfaces for cubemaps) */
+    const UnsignedInt numImages = (isCubemap) ? 6 : 1;
+    for (UnsignedInt n = 0; n < numImages; ++n) {
+        if (compressed) {
+            loadCompressedImageData(colorFormat.compressed, width, height, depth, dimensions);
+        } else {
+            loadUncompressedImageData(colorFormat.uncompressed, width, height, depth, components, dimensions);
+        }
+
+        UnsignedInt w = width;
+        UnsignedInt h = height;
+        UnsignedInt d = depth;
+
+        /* load all mipmaps for current surface */
+        for (UnsignedInt i = 0; i < numMipmaps && (w || h); i++) {
+            /* shrink to next power of 2 */
+            w = Math::max(w >> 1, 1u);
+            h = Math::max(h >> 1, 1u);
+            d = Math::max(d >> 1, 1u);
```

Comment:
Just a suggestion, but I think that using `Vector3i`/`Vector3ui` instead of separate width/height/depth would reduce the verbosity and amount of function parameters quite a bit. Math functions will work without change (as expected), including `Math::max()` and operator `>>`.